### DISCUSSION TITLE
Error trying to load a model with OMPython

### DISCUSSION THREAD

**Discussion by ejperezHI** [OpenModelica/OpenModelica#12060] [Q&A]
Hello,
I am trying to load and execute an OpenModelica model as always with this code:
`omc = OMCSessionZMQ()`
`mod = ModelicaSystem(modelpath, name_fichero)`

And the error message I receive is this one:
<img width="1052" alt="Captura de pantalla 2024-03-04 a las 11 21 35" src="https://github.com/OpenModelica/OpenModelica/assets/131266350/60f5a1d3-8609-4576-926e-dfa57f66f8af">

It seems that instead of reading my direction for opening my OpenModelica model, it is reading that direction to open a library?

This simple code and model has always worked for me well (last day I tried was two days ago) , so I don't know where is the issue and why is it.

Thanks a lot in advance,

**Comment by casella** [comment]
@arun3688 could you please check when you have some time?

**Comment by adeas31** [comment]
We need more information. What version of OpenModelica and OMPython you are using? Also, is `name_fichero` using the package `motor_completo`?

**Reply by ejperezHI** [reply]
Hello and thanks for the reply,
The point is that executing those commands written above in my local machine with OMPython 3.4.0 works perfectly, but with 3.5.0 it does not.

Also, executing those commands written above in an external machine it gives the same error (both with OMPython 3.4.0 or 3.5.0)

Thanks!

**Reply by arun3688** [reply]
@ejperezHI can you upload your model and let me see if i can reproduce the issue, because it should work by default and your error message is not clear, can you post the full error message

**Reply by ejperezHI** [reply]
Thanks for the reply,

Here it goes.

The Error message is:
`Error: Failed to load package motor_completo (default) using MODELICAPATH [/Users/ejperezhi/.openmodelica/libraries/.](https://file+.vscode-resource.vscode-cdn.net/Users/ejperezhi/.openmodelica/libraries/)`
`Error: Class motor_completo not found in scope <top>.`

I cannot attach de model in this message, it gives me error that "we don't support that file type"

**Reply by adeas31** [reply]
You can attach as .zip or .txt file.

**Reply by ejperezHI** [reply]
Thanks!
Here goes
[motor_completo.mo.zip](https://github.com/OpenModelica/OpenModelica/files/14523394/motor_completo.mo.zip)

**Comment by casella** [comment]
Maybe related to https://github.com/OpenModelica/OMPython/issues/199?

**Reply by ejperezHI** [reply]
Thanks for your reply,
I already thougth that and already checked that issue, but no. It seems that OpenModelica is trying to open my .mo model as a library instead as a .mo model

It works perfectly fine in my local machine with OMPython 3.4.0.

It has the issue that I reported here in my local machine with OMPython 3.5.0 and in an external machine (linux based) with OMPython 3.4.0

**Comment by arun3688** [comment]
@ejperezHI I have updated the OMPython please update your OMPython package via `pip install OMPython -U`, but i am sure your issue is related with docker and the Modelica libraries are not installed in the needed place. But for me it worked fine 

```
>>> from OMPython import ModelicaSystem
>>> mod = ModelicaSystem("C:/OPENMODELICAGIT/buildingstest/motor_completo/motor_completo.mo", "motor_completo", verbose=False)
>>> mod.simulate()
LOG_SUCCESS       | info    | The initialization finished successfully without homotopy method.
LOG_SUCCESS       | info    | The simulation finished successfully.
```

**Reply by ejperezHI** [reply]
Thank you for the reply!
I have the correct libraries installed (I checked it out).

The mystery here is that if I put wrong the direction of the model in the first argument of "ModelicaSystem()", it shows me an error telling me that it has checked in the directory I want it to check and didn't found that model.

However, if I put the path correctly, it gives me the error posted above of "failed to load package using modelicapath"
